crates/vhost/src/vhost_user/slave_req_handler.rs

```diff
@@ -432,6 +432,9 @@ impl<S: VhostUserSlaveReqHandler> SlaveReqHandler<S> {
                 self.acked_protocol_features = msg.value;
                 self.update_reply_ack_flag();
                 self.send_ack_message(&hdr, res)?;
+
+                #[cfg(feature = "xen")]
+                self.check_proto_feature(VhostUserProtocolFeatures::XEN_MMAP)?;
```

Comment:
Okay, so here we always fail if `xen` feature is enabled, but `XEN_MAP` not negotiated. I think we should document this, but in general I'd prefer that the presence of a feature or not is handled at runtime if it's not too complicated.

Reply:
With `if cfg!()` ? Or something else ? Though since we used the compile time option everywhere in vm-memory crate, it made sense to do the same here since both can't be used at once.

Reply:
Yep, this is fine now I get how the xen support works ;-)